### DISCUSSION TITLE
【教程】一文解决如何使用gocq的dev

### DISCUSSION THREAD

**Discussion by KomeiDiSanXian** [Mrs4s/go-cqhttp#2420] [General]
### 写在前面
目前，看到的一些issue是有关签名服务器和滑块验证码不知道怎么过的，而使用 `Relases v1.1.0` 存在一些问题（比如使用不兼容的qsign版本）使用dev分支目前看来是较好的选择

但是dev分支对于 `config.yml` 做出了更新，使用旧版本的`config.yml` 会出现各种问题，加上文档似乎删掉了如何手动获取ticket的内容，所以本文将带你从头开始，对 **dev** 版本的go-cqhttp进行配置和启动

综上，本文可以说只是对 [文档启动部分](https://docs.go-cqhttp.org/guide/quick_start.html#%E5%9F%BA%E7%A1%80%E6%95%99%E7%A8%8B) 进行一些补充说明

**注意：** 顾名思义，dev就是正在开发的版本，如果你在使用途中遇到了bug，提交issue时请勾选`我已经使用 dev分支版本 测试过，问题依旧存在。`如果可以，指明所使用的 `commitID`，就是类似 `ff236ad` 这样的字符串

碎碎念：什么时候更新文档

---
### 对于老版本
commit f8354ec0826f14131f4600f52600d2cc6c42cc76 对`config.yml`做出了很大的修改，所以在此之前的版本最好删掉 `config.yml` 重新生成一份

---
### 下载
如果你会自己构建了，我觉得你可以跳过这一步了
如果你不会，最简单的方法就是 [点我](https://github.com/Mrs4s/go-cqhttp/actions/workflows/ci.yml) 然后按照以下步骤：

1. 选择最上面的，且被蓝色包围的 `dev` 所对应的网页
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/2c487856-d3ed-4591-b4dc-509f28bead18)

2. 选择可以在你系统上运行的版本下载
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/36608446-77d1-4e3e-96ad-dca35faec778)

3. 解压下载的文件到一个你喜欢的位置
4. （可选） [点我](https://github.com/fuqiuluo/unidbg-fetch-qsign/releases) 去下载qsign，推荐使用 **1.1.6及以上版本**
> 对于qisgn的部署，可以参考[这里](https://github.com/fuqiuluo/unidbg-fetch-qsign/wiki)，本文仅做出一些补充说明

---
### 初次启动
>注意：在 windows 系统启动需要使用的是shell，双击后请点击 **是** 
>在这以后需要启动就运行 `go-cqhttp.bat`
>推荐你使用`cmd`或`powershell`启动来跳过这一步骤
1. 进入到命令行以后，按需选择通信方式
2. 退出后，编辑`config.yml`

---
### 有关 config.yml 的签名服务器部分
本文不再赘述账号设置等步骤，只阐述配置qsign签名服务器时容易踩的坑和推荐做出的一些操作
> config.yml 注释里写得很清楚，请根据你自己的需求按需修改，如果还是不懂欢迎提问
1. qsign是基于**安卓**平台进行计算的，所以使用其他的协议配合使用qsign是有奇奇怪怪的问题的，所以在gocq的协议尽量就不要改了，默认的是 `Android Pad 8.9.63.11390`
2. qsign对于`basePath`的选择推荐是`txlib/8.9.63`，对应gocq的协议，更高版本的已经有人踩坑被冻结，目前看来是效果因号而异（？
3. **强烈推荐**开启qsign的自动注册功能，也就是修改 `txlib/[你使用的协议版本]/config.json` 中的 `auto_register`为true，gocq的auto-register由于实现问题建议设为false
> 这样的好处是qsign崩了就无需再次启动gocq
4. 如果你使用的是第三方提供的qsign，请注意修改 `config.yml` 签名服务器部分的key和authorization，具体可以去询问第三方，对应不上就不能使用

---
### 运行
编辑完`config.yml`后，可以参照以下步骤，若不出意外，你的命令行应该类似如下图片
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/1d08caa8-a1bd-47fc-9a03-e81b2f80acbd)

如果不能使用自动提交，请选 2 并回车，在浏览器键入所给的地址
**以下内容仅针对如何进行手动提交ticket**
以chrome浏览器抓取ticket为例
1. 打开chorme后按下f12
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/74ec2fb8-59f1-476e-8b27-2af79a99e1a8)
2. 点击`Network`或者`网络`（由你的语言所决定）
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/c5515a2d-6933-490c-be91-2ffd9c880e9f)
3. 地址栏键入gocq给你的网址
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/0448ee52-11d2-4979-9b18-cbda1a42623a)
4. 滑动滑块，跳转到下一个页面，如果还有验证，继续验证
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/cf933b17-9c4c-496a-a345-4efae14aef35)
5. 重复步骤4获得 `cap_union_new_verify`，如图
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/08d390fc-442e-4537-92c2-8c70033d5215)
6. 点击 `cap_union_new_verify` 后，点击 `Preview`，如图
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/6fb051b6-3625-4c40-9d41-ec7a3c38bf37)
7. 复制 ticket （就是那一长串字符）粘贴到 gocq里并回车，注意一定要把引号里的内容复制全了，引号不要
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/c2bc8350-7b53-4a9b-a665-355038e0a37f)

> 没有清除那堆花花绿绿的东西找不到 cap_union_new_verify？没关系，参照这张图就可以找到
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/db1d81a6-cb2d-476e-b8a4-6e590ea5ab6a)

接下来按照gocq的指示进行就可以了

---
### 其他
目前看来自动提交已经恢复了，至少我写的时候是可以了
教学大概是差不多了，如果还有什么问题的话可以去issue和discussion寻找，会有人解答的（

**Comment by Clyde-Li** [comment]
请问authorization字段在qsign里对应哪一项？本地搭的1.1.9用的8.9.63，在config里只找到了key

**Reply by KomeiDiSanXian** [reply]
![图片](https://github.com/Mrs4s/go-cqhttp/assets/53388791/ebfdf90b-05a7-49ff-8f96-739f86be63bc)

`key`下面的`authorization`是会自动生成的，这个是在gocq的`config.yml`里填写的，不在qsign里

**Comment by jinyu2022** [comment]
啥时候合并进master啊，要等到1.2.0？

**Comment by Little1Yuan** [comment]
#2461 一个也打不开

**Comment by PackageInstaller** [comment]
那么，怎么换更高的版本呢，我qsign用的8.9.73

**Comment by RandomLemon** [comment]
今日code45了，R.I.P.
